code/game/objects/structures/fuel_port.dm

```diff
@@ -0,0 +1,82 @@
+/obj/structure/fuel_port
+	name = "fuel port"
+	desc = "The fuel input port of the shuttle. Holds one fuel tank. Use a crowbar to open and close it."
+	icon = 'icons/obj/structures/fuel_port.dmi'
+	icon_state = "base"
+
+	density = FALSE
+	anchored = TRUE
+
+	var/open = FALSE
+	var/parent_shuttle
+
+	var/sound_open = 'sound/effects/locker_open.ogg'
+	var/sound_close = 'sound/effects/locker_close.ogg'
+
+	/// Used to create a prepared tank on initialization.
+	var/start_tank_type
+
+/obj/structure/fuel_port/Initialize()
+	. = ..()
+	if(start_tank_type)
+		new start_tank_type(src)
+
+/obj/structure/fuel_port/proc/locate_tank()
+	return locate(/obj/item/tank) in contents
+
+/obj/structure/fuel_port/attack_hand(mob/user)
+	if(!open)
+		to_chat(user, SPAN_WARNING("The door is secured tightly. You'll need a crowbar to open it."))
+		return
+	else
+		var/obj/item/tank/tank = locate_tank()
+		if(tank)
+			user.put_in_hands(tank)
+
+	update_icon()
+
+/obj/structure/fuel_port/on_update_icon()
+	cut_overlays()
+
+	if(open)
+		add_overlay(mutable_appearance(icon, "[icon_state]_open"))
+		var/obj/item/tank/tank = locate_tank()
+		if(tank)
+			if(tank.color)
+				add_overlay(mutable_appearance(icon, "[icon_state]_tank", tank.color))
+			else
+				add_overlay(mutable_appearance(icon, "[icon_state]_tank_orange"))
+	else
+		add_overlay(mutable_appearance(icon, "[icon_state]_closed"))
+
+/obj/structure/fuel_port/attackby(obj/item/W, mob/user)
+	if(isCrowbar(W))
```

Comment:
Please use the new tool interaction proc if you're adding new code that uses tools.